ext/standard/proc_open.c

```diff
@@ -381,22 +385,18 @@ static inline HANDLE dup_fd_as_handle(int fd)
 # define close_descriptor(fd)	close(fd)
 #endif
 
-#define DESC_PIPE		1
-#define DESC_FILE		2
-#define DESC_REDIRECT	3
-#define DESC_PARENT_MODE_WRITE	8
-
 struct php_proc_open_descriptor_item {
-	int index; 							/* desired fd number in child process */
-	php_file_descriptor_t parentend, childend;	/* fds for pipes in parent/child */
-	int mode;							/* mode for proc_open code */
-	int mode_flags;						/* mode flags for opening fds */
+	int index;                                 /* desired FD number in child process */
+	int is_pipe;
+	php_file_descriptor_t parentend, childend; /* FDs for pipes in parent/child */
+	int mode_flags;                            /* mode flags for opening FDs */
 };
 /* }}} */
 
 static zend_string *get_valid_arg_string(zval *zv, int elem_num) {
 	zend_string *str = zval_get_string(zv);
 	if (!str) {
+		zend_value_error("Command array element %d could not be converted to string", elem_num);
```

Comment:
This shouldn't be here, `zval_get_string` already throws.

Reply:
Thanks for the catch.